src/utils/date.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * Convert time zone-aware date/time string to Pacific/Auckland time zone string
+ *
+ * @param dateTimeString Optional date/time string which can be parsed by the `Date` constructor
+ * @returns Localised date/time string eg "2024-01-01"
+ *
+ */
+export function getPacificAucklandYearMonthDay(dateTimeString?: string | null): string | undefined {
+  if (dateTimeString == null) return;
+
+  // "sv-SE" formats date times as "yyyy-MM-dd hh:mm:ss"
+  const pacificAucklandDateTimeString = new Date(dateTimeString).toLocaleString('sv-SE', {
```

Comment:
Do we want to use [`toISOString`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString)? I don't really like depending on a specific locale, especially one which isn't local.

Reply:
`getPacificAucklandYearMonthDay` is used by the [mapsheet-coverage command](https://github.com/linz/argo-tasks/blob/326adaf65094477536136b4cadb4a0cd2fcea8ca/src/commands/mapsheet-coverage/mapsheet.coverage.ts#L173). I've moved it in this PR because there was no need for date-related code in `generate-path` any more. @blacha could comment?

Reply:
we cannot use `toISOString` as "The timezone is always UTC"

this formats the string as a almost ISO while putting the timezone as Pacific/Auckland.